controllers/storagecluster/cephfilesystem.go

```diff
@@ -121,11 +124,79 @@ func (obj *ocsCephFilesystems) ensureCreated(r *StorageClusterReconciler, instan
 				return reconcile.Result{}, err
 			}
 		}
+		// create default csi subvolumegroup for the filesystem
+		// skip for the ocs provider mode
+		if !instance.Spec.AllowRemoteStorageConsumers {
+			err = r.createDefaultSubvolumeGroup(cephFilesystem.Name, cephFilesystem.Namespace, cephFilesystem.ObjectMeta.OwnerReferences)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+		}
 	}
 
 	return reconcile.Result{}, nil
 }
 
+func (r *StorageClusterReconciler) createDefaultSubvolumeGroup(filesystemName, filesystemNamespace string, ownerReferences []metav1.OwnerReference) error {
+	// TODO: After fix of rook issue https://github.com/rook/rook/issues/13220 add svg name spec
+
+	existingsvg := &cephv1.CephFilesystemSubVolumeGroup{}
+	err := r.Client.Get(r.ctx, types.NamespacedName{Name: defaultSubvolumeGroupName, Namespace: filesystemNamespace}, existingsvg)
+	if err == nil {
+		if existingsvg.DeletionTimestamp != nil {
+			r.Log.Info("Unable to restore subvolumegroup because it is marked for deletion.", "subvolumegroup", klog.KRef(filesystemNamespace, defaultSubvolumeGroupName))
+			return fmt.Errorf("failed to restore subvolumegroup %s because it is marked for deletion", existingsvg.Name)
+		}
+	}
+
+	objectMeta := metav1.ObjectMeta{Name: defaultSubvolumeGroupName, Namespace: filesystemNamespace, OwnerReferences: ownerReferences}
+	cephFilesystemSubVolumeGroup := &cephv1.CephFilesystemSubVolumeGroup{ObjectMeta: objectMeta}
+	mutateFn := func() error {
+		cephFilesystemSubVolumeGroup.Spec = cephv1.CephFilesystemSubVolumeGroupSpec{
+			FilesystemName: filesystemName,
+		}
+		return nil
+	}
+	_, err = ctrl.CreateOrUpdate(r.ctx, r.Client, cephFilesystemSubVolumeGroup, mutateFn)
+	if err != nil {
+		r.Log.Error(err, "Could not create/update default csi cephFilesystemSubVolumeGroup.", "cephFilesystemSubVolumeGroup", klog.KRef(cephFilesystemSubVolumeGroup.Namespace, cephFilesystemSubVolumeGroup.Name))
+		return err
+	}
+	return nil
+}
+
+func (r *StorageClusterReconciler) deleteDefaultSubvolumeGroup(filesystemName, filesystemNamespace string, ownerReferences []metav1.OwnerReference) error {
+	existingsvg := &cephv1.CephFilesystemSubVolumeGroup{}
+	err := r.Client.Get(r.ctx, types.NamespacedName{Name: defaultSubvolumeGroupName, Namespace: filesystemNamespace}, existingsvg)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			r.Log.Info("Uninstall: csi subvolumegroup not found.", "Subvolumegroup", klog.KRef(filesystemNamespace, defaultSubvolumeGroupName))
+			return nil
+		}
+		r.Log.Error(err, "Uninstall: Unable to retrieve subvolumegroup.", "subvolumegroup", klog.KRef(filesystemNamespace, defaultSubvolumeGroupName))
+		return fmt.Errorf("uninstall: Unable to retrieve csi subvolumegroup : %v", err)
+	}
+
+	if existingsvg.GetDeletionTimestamp().IsZero() {
```

Comment:
@iamniting Can you confirm the timestamp part
I used a similar one of the filesystem.

when it sets zero `.IsZero()`

Reply:
IsZero to true means it is nil and not marked for deletion.